test/Npgsql.Tests/LoadBalancerTests.cs

```diff
@@ -0,0 +1,175 @@
+        createCluster();
+
+        try
+        {
+            conns = CreateConnections(connStringBuilder, numConns);
+            await VerifyOn("127.0.0.1", numConns/3);
+            await VerifyOn("127.0.0.2", numConns/3);
+            await VerifyOn("127.0.0.3", numConns / 3);
+        }
+        catch (Exception ex)
+        {
+            Console.WriteLine("Failure:" + ex.Message);
+            Console.WriteLine("Failure stacktrace: " + ex.StackTrace);
+        }
+        finally
+        {
+            foreach (var conn in conns)
+            {
+                conn.Close();
+            }
+            DestroyCluster();
+        }
+    }
+
+    [Test]
+    public async Task TestLoadBalance2()
+        {
+            var connStringBuilder = "host=127.0.0.1;port=5433;database=yugabyte;userid=yugabyte;password=yugsbyte;Load Balance Hosts=true;YB Servers Refresh Interval=30;Timeout=0";
+            List<NpgsqlConnection> conns = new List<NpgsqlConnection>();
+
+            createCluster();
+            try
+            {
+                var conn1 = CreateConnections(connStringBuilder, numConns);
+                conns.AddRange(conn1);
+
+                string? _Output = null;
+                string? _Error = null;
+                var cmd = "/bin/yb-ctl stop_node 1";
+                ExecuteShellCommand(cmd, ref _Output, ref _Error );
+                Console.WriteLine(_Output);
+
+                System.Threading.Thread.Sleep(30000);
+
+                var conn2 = CreateConnections(connStringBuilder, numConns);
+                conns.AddRange(conn2);
+
+                await VerifyOn("127.0.0.2", 5);
+                await VerifyOn("127.0.0.3", 5);
+            }
+            catch (Exception ex)
+            {
+                Console.WriteLine("Failure:" + ex.Message);
+                Console.WriteLine("Failure stacktrace: " + ex.StackTrace);
+            }
+            finally
+            {
+                foreach (var conn in conns)
+                {
+                    if (conn.State != System.Data.ConnectionState.Closed)
+                    {
+                        conn.Close();
+                    }
+                }
+                DestroyCluster();
+            }
+        }
+
+    [Test]
+    public async Task TestLoadBalance3()
+    {
+        var connStringBuilder = "host=127.0.0.1;port=5433;database=yugabyte;userid=yugabyte;password=yugsbyte;Load Balance Hosts=true;Timeout=0";
+
+        List<NpgsqlConnection> conns = new List<NpgsqlConnection>();
+        createCluster();
+        try
+        {
+            List<Thread> threads = new List<Thread>();
+            var numThreads = 15;
+            for (var i = 0; i < numThreads; i++)
+            {
+                List<NpgsqlConnection> conn = new List<NpgsqlConnection>();
+                Thread thread = new Thread(() => { conn = CreateConnections(connStringBuilder, numConns); });
+                threads.Add(thread);
+                thread.Start();
```

Comment:
Please invoke the `thread.Start()` together for all threads but in a separate loop.
```
foreach (var thread in threads) {
  thread.Start();
}
```